internal/vxproto/proto.go

```diff
@@ -854,12 +889,139 @@ func (vxp *vxProto) GetIMCModule(token string) IModuleSocket {
 	return nil
 }
 
+// GetIMCTopic is function which get topic info for imc topic token
+// The function may return nil if the topic token doesn't exist
+func (vxp *vxProto) GetIMCTopic(topic string) ITopicInfo {
+	if !vxp.HasIMCTopicFormat(topic) {
+		return nil
+	}
+
+	vxp.mutex.RLock()
+	defer vxp.mutex.RUnlock()
+
+	if info, ok := vxp.topics[topic]; ok && info != nil {
+		return info
+	}
+
+	return nil
+}
+
 // MakeIMCToken is function which get new imc token
 func (vxp *vxProto) MakeIMCToken(name, gid string) string {
 	hash := md5.Sum(append([]byte(gid+":"+name+":"), vxp.tokenKey...))
 	return "ffffffff" + hex.EncodeToString(hash[:])
 }
 
+// MakeIMCTopic is function which get new imc topic
+func (vxp *vxProto) MakeIMCTopic(name, gid string) string {
+	hash := md5.Sum(append([]byte(gid+":"+name+":"), vxp.tokenKey...))
+	return "ffff7777" + hex.EncodeToString(hash[:])
+}
+
+// SubscribeIMCToTopic is function to make or extend record about topic in vxproto
+func (vxp *vxProto) SubscribeIMCToTopic(name, gid, token string) bool {
+	if !vxp.HasIMCTokenFormat(token) {
+		return false
+	}
+
+	vxp.mutex.Lock()
+	defer vxp.mutex.Unlock()
+
+	topic := vxp.MakeIMCTopic(name, gid)
+	if info, ok := vxp.topics[topic]; ok && info != nil {
+		if !stringInSlice(token, info.subscriptions) {
+			info.subscriptions = append(info.subscriptions, token)
+		}
+	} else {
+		vxp.topics[topic] = &topicInfo{
+			name:          name,
+			groupID:       gid,
+			subscriptions: []string{token},
+		}
+	}
+
+	return true
+}
+
+// UnsubscribeIMCFromTopic is function to remove record about topic in vxproto
+func (vxp *vxProto) UnsubscribeIMCFromTopic(name, gid, token string) bool {
+	if !vxp.HasIMCTokenFormat(token) {
+		return false
+	}
+
+	vxp.mutex.Lock()
+	defer vxp.mutex.Unlock()
+
+	topic := vxp.MakeIMCTopic(name, gid)
+	if info, ok := vxp.topics[topic]; ok && info != nil {
+		subs := info.subscriptions
+		for idx, sub := range subs {
```

Comment:
Maybe it is better to use "set" (map[string]struct{}) for subscriptions, such a change can increase readability of this code as it will be changed to something like:
```
if info, ok := vxp.topics[topic]; ok && info != nil {
  delete(info.subscriptions, token)
}
```

Reply:
it's not about security/performance - but only readability